internal/cli/atlas/serverless/backup/restores/watch.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2023 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package restores
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli"
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli/require"
+	"github.com/mongodb/mongodb-atlas-cli/internal/config"
+	"github.com/mongodb/mongodb-atlas-cli/internal/flag"
+	"github.com/mongodb/mongodb-atlas-cli/internal/store"
+	"github.com/mongodb/mongodb-atlas-cli/internal/usage"
+	"github.com/spf13/cobra"
+)
+
+type WatchOpts struct {
+	cli.GlobalOpts
+	cli.WatchOpts
+	id          string
+	clusterName string
+	store       store.ServerlessRestoreJobsDescriber
+}
+
+func (opts *WatchOpts) initStore(ctx context.Context) func() error {
+	return func() error {
+		var err error
+		opts.store, err = store.New(store.AuthenticatedPreset(config.Default()), store.WithContext(ctx))
+		return err
+	}
+}
+
+func (opts *WatchOpts) watcher() (bool, error) {
+	result, err := opts.store.ServerlessRestoreJob(opts.ConfigProjectID(), opts.clusterName, opts.id)
+	if err != nil {
+		return false, err
+	}
+	return result.Expired || result.Cancelled || *result.Failed || result.FinishedAt != "", nil
+}
+
+func (opts *WatchOpts) Run() error {
+	if err := opts.Watch(opts.watcher); err != nil {
+		return err
+	}
+
+	return opts.Print(nil)
```

Comment:
Unix convention is for watch command to return different return code. For failed it can be 1 for success it can be zero. This will enable scripting/ci/cd cases. For example this is community build script that tries to use watch (with some issues) 

![Screenshot 2023-03-15 at 10 38 53](https://user-images.githubusercontent.com/981838/225269482-2f38d3ce-e522-436d-9fb9-9909c9c397f4.png)

Reply:
This is exactly what is happening here `return opts.Print(nil)` returns nil and consequently, the exit code is 0. 
The `return err` on the other hand will return non-zero.

Reply:
[q] https://github.com/mongodb/mongodb-atlas-cli/pull/1765/files#r1136788922 returns error as nil. 
Does opts.Watcher return error based on boolean value?